pkg/kiln/docker_registry.go

```diff
@@ -0,0 +1,145 @@
+package kiln
+
+import (
+	"fmt"
+
+	"github.com/30x/kiln/pkg/registry"
+	"github.com/docker/engine-api/types"
+)
+
+//RegistryImageCreator is a struct that holds our pointer to the docker client
+type RegistryImageCreator struct {
+	//the client to docker
+	client registry.Registry
+	//the remote repository url
+	dockerCreator ImageCreator
+}
+
+//NewRegistryImageCreator creates an instance of the RegistryImageCreator from the docker environment variables, and returns the instance
+func NewRegistryImageCreator(repo string, reg registry.Registry) (ImageCreator, error) {
+	// necessary for using a GCR registry, the project name needs to be in the image tag
+	// and tagging is done by the LocalImageCreator before push
+	if proj := reg.GetProjectName(); proj != "" {
```

Comment:
Are you proposing splitting this implementation into two. One for the generic and another for GCR to avoid the if() statements? That seems neater but then we have another two implementations without much differences.

I'd say leave for now until it become a bigger problem.